core/sql/langman/LmRoutineJavaObj.cpp

```diff
@@ -192,6 +192,7 @@ LmResult LmRoutineJavaObj::invokeRoutineMethod(
           (*emitRowPtr_)(NULL,0,&qs);
         }
     }
+  jni->DeleteLocalRef(jniResult);
```

Comment:
First, sorry for having caused this memory leak and thank you for fixing it!

One question: The code saves data members of the jniResult object in returnedIIL_ and returnedPIL_. Is it legal to destroy the local ref to the Java ReturnInfo object and to keep using local refs to the strings and byte arrays stored in it? If it's not, or if it's questionable, then we could save this local ref in the LmRoutineJavaObj object and clean it up on the next call or in the destructor, like we do it for returnedIIL_. If you like I'd be happy to code this and send you a patch.